### PR TITLE
waldo-upload 0.2.0

### PR DESCRIPTION
**Pull request by eBardX**
TagCheck

### What to do if the build fails?

At the moment contributors do not have access to the CI workflow triggered by StepLib PRs. In case of a failed build, we ask for your patience. Maintainers of Bitrise Steplib will sort it out for you or inform you if any further action is needed.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

**New Step**
Thank you for the new Step share! The CI check might will fail due to our extended validation engine. Nothing to worry about yet, we will get back to you shortly.

### CODE REVIEW

File: steps/waldo-upload/0.2.0/step.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    summary: A `.app`, `.ipa`, or `.apk` path.
+    title: Build path
+- opts:
+    description: |
+      Each application registered in Waldo is associated with a unique upload
+      token.
+    is_required: true
+    is_sensitive: true
+    title: Waldo upload token
+  upload_token: $WALDO_UPLOAD_TOKEN
```
Comment (koral--) [inline steps/waldo-upload/0.2.0/step.yml:60]: Only env vars exported by bitrise.io, CLI, other steps or shell should be used as default values. Not custom ones.

File: steps/waldo-upload/0.2.0/step.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+- include_symbols: "no"
+  opts:
+    category: Symbols
+    description: |
+      **iOS only.** Attempts to find the symbols associated with the
+      specified `.app` build path.
+
+      This input is _ignored_ if a symbols path is specified.
+    is_required: false
+    title: Include symbols with the build upload
```
Comment (koral--) [inline steps/waldo-upload/0.2.0/step.yml:70]: This title is misleading for me. Taking description into account it should be rather called like "Search for symbols". It sounds preposterous that "including symbols" is ignored when symbols path is specified.

File: steps/waldo-upload/0.2.0/step.yml (hunk 1 above)
Comment (eBardX) [reply steps/waldo-upload/0.2.0/step.yml:60]: Yeah, that makes sense in this context. I'll remove it.

File: steps/waldo-upload/0.2.0/step.yml (hunk 2 above)
Comment (eBardX) [reply steps/waldo-upload/0.2.0/step.yml:70]: I took the name from the underlying script that this step is wrapping. Normally, when you build an app for simulator, Xcode leaves the `.dSYM` bundle right next to the `.app` bundle. In that case, `include_symbols` makes sense. Unfortunately, the Bitrise `Xcode build for simulator` step _copies_ the `.app` bundle from some internal location to a more public one (pointed to by `$BITRISE_APP_DIR_PATH`) but does _not_ do the same for the `.dSYM` bundle (and thus `$BITRISE_DSYM_PATH` is useless). I used `include_symbols` to name this input in the hope that eventually the `Xcode build for simulator` step is fixed.

If you insist, I can rename the input to `find_symbols` or something similar.

**Comment by eBardX** [comment]
Do you have any idea why the `tooling-steplib-controller` check is failing. The `Details` link takes me to a 404 page. 😞

**Pull request closed without merging**